programs/stake/src/stake_state.rs

```diff
@@ -731,6 +756,20 @@ impl Stake {
             Ok(())
         }
     }
+
+    fn can_merge_with(&self, source: &Self) -> Result<(), InstructionError> {
```

Comment:
I still need to figure out something better for the `can_merge_with()` helpers.  For now, I made it worse and added one to `Meta` :upside_down_face: